### PR TITLE
Fix naming: `podTemplatePodSecurityContext`

### PR DESCRIPTION
**Pull request by stone-z**
## Description

https://github.com/aquasecurity/trivy-operator/pull/477 made the securityContext of containers within a scan job configurable (:pray:) and renamed `scanJobPodTemplateSecurityContext` to `scanJobPodTemplatePodSecurityContext` to differentiate the option from the new `scanJobPodTemplateContainerSecurityContext` option.

However, some instances of the old name were left behind, meaning that the configuration set via Helm values was never actually used in scan Jobs.

This PR finishes the renaming so that the PodSecurityContext is read correctly from the ConfigMap.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@nerzhul  @FelixKampfer  FYA.

**Comment by chen-keinan** [comment]
@stone-z thanks for the contribution LGTM 🚀  once all checks are passing we are good for merge

**Comment by stone-z** [comment]
Ah I missed renaming the value in the tests. Should pass now

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**
